### PR TITLE
OFS-205: Validation for unique PH display name

### PR DESCRIPTION
**Pull request by dborowiecki**
TICKET: [OFS-205](https://openimis.atlassian.net/browse/OFS-205)

### CODE REVIEW

File: policyholder/gql/gql_mutations/create_mutations.py [hunk 1]
```diff
@@ -12,6 +13,11 @@ class CreatePolicyHolderMutation(BaseHistoryModelCreateMutationMixin, BaseMutati
     class Input(PolicyHolderInputType):
         pass
```
Comment (sniedzielski) [inline policyholder/gql/gql_mutations/create_mutations.py:15]: I think we have to override the _mutate method from core module here and add to that method after 'pop client_mutation' data "PolicyHolderValidation.validate_create(user, **data)"
example:
```
@classmethod
    def _mutate(cls, user, **data):
        if "client_mutation_id" in data:
            data.pop('client_mutation_id')
        if "client_mutation_label" in data:
            data.pop('client_mutation_label')
        PolicyHolderValidation.validate_create(user, **data)
        cls.create_object(user=user, object_data=data)
```

File: policyholder/gql/gql_mutations/update_mutations.py [hunk 2]
```diff
@@ -12,6 +13,11 @@ class UpdatePolicyHolderMutation(BaseHistoryModelUpdateMutationMixin, BaseMutati
     class Input(PolicyHolderUpdateInputType):
         pass
```
Comment (sniedzielski) [inline policyholder/gql/gql_mutations/update_mutations.py:15]: I think we have to override the _mutate method from core module here and add to that method after 'pop client_mutation' data "PolicyHolderValidation.validate_update(user, **data)"
example:
```
@classmethod
    def _mutate(cls, user, **data):
        if "client_mutation_id" in data:
            data.pop('client_mutation_id')
        if "client_mutation_label" in data:
            data.pop('client_mutation_label')
        PolicyHolderValidation.validate_update(user, **data)
        updated_object = cls._model.objects.filter(id=data['id']).first()
        [setattr(updated_object, key, data[key]) for key in data]
        cls.update_object(user=user, object_to_update=updated_object)
```

**Comment by sniedzielski** [comment]
@dborowiecki Looks really good. Please relate to the comment - I think that we only have to override the "_mutate" method. Then we can merge after solving this issue.

File: policyholder/gql/gql_mutations/create_mutations.py (hunk 1 above)
Comment (dborowiecki) [reply policyholder/gql/gql_mutations/create_mutations.py:15]: Validation is called inside async_mutate in [BaseMutation](https://github.com/openimis/openimis-be-core_py/blob/c30a49acb2057c94620cf93253773dbb32ed62e8/core/gql/gql_mutations/base_mutation.py#L21) class. There is no need for overriding _mutate

File: policyholder/gql/gql_mutations/update_mutations.py (hunk 2 above)
Comment (dborowiecki) [reply policyholder/gql/gql_mutations/update_mutations.py:15]: Same as for insert, validation is performed in base class

File: policyholder/gql/gql_mutations/create_mutations.py (hunk 1 above)
Comment (sniedzielski) [reply policyholder/gql/gql_mutations/create_mutations.py:15]: ok

File: policyholder/gql/gql_mutations/update_mutations.py (hunk 2 above)
Comment (sniedzielski) [reply policyholder/gql/gql_mutations/update_mutations.py:15]: ok

**Review by sniedzielski — APPROVED** [review]

**Review by delcroip — APPROVED** [review]

**Pull request merged**